starfish/image/_stack.py

```diff
@@ -104,16 +105,16 @@ def tile_opener(toc_path, tile, ext):
             tile_opener=tile_opener)
 
     def max_proj(self, dim):
-        valid_dims = ['hyb', 'ch', 'z']
+        valid_dims = [Keys.HYB, Keys.CH, Keys.ZLAYER]
```

Comment:
Does this assume the user will request the dim by its name? Can we also support the numeric value of the axis?

Reply:
@dganguli I'm interested in your thoughts here. I see Tony's point, but I'm wondering what you think users will think about having to use names only.

Reply:
i personally love it. i'm not a fan of keeping track of numeric values of the axes -- it leads to a lot of painful bugs for the user (e.g., me when i first started working on this) and makes it hard to reorder axes internally (which i also struggled with initially). but i can be convinced otherwise 😀 i think people can always get the numpy array if they want to just deal with that too right?

Reply:
Ah, fair point. Ok, I'm fine with this then.